core/modules/commands/server.js

```diff
@@ -141,7 +139,11 @@ SimpleServer.prototype.listen = function(port,host) {
 				});
 				break;
 		}
-	}).listen(port,host);
+	};
+	
+SimpleServer.prototype.listen = function(port,host) {
+	var self = this;
+	http.createServer(self.requestHandler.bind(null,self)).listen(port,host);
```

Comment:
There is no inner scope so `this` works without `self`. Also if `requestHandler` never uses `this` then there is no need to `bind` Also it removes the need to be on the `SimpleServer.prototype`.

What if requestHandler dropped the `self` and used `this`?

```js
http.createServer(this.requestHandler.bind(this)).listen(port,host);
```